app/presenters/reservation_calendar.rb

```diff
@@ -0,0 +1,37 @@
+class ReservationCalendar < SimpleDelegator
+
+  attr_accessor :hostname
+
+  def self.to_calendar(reservation, hostname)
+    new(reservation, hostname).as_ical
+  end
+
+  def initialize(reservation, hostname)
+    super(reservation)
+    @hostname = hostname
+  end
+
+  def as_ical
+    cal = Icalendar::Calendar.new
+    cal.event do |e|
+      e.dtstart = Icalendar::Values::DateTime.new(reserve_start_at)
+      e.dtend = Icalendar::Values::DateTime.new(reserve_end_at)
+      e.summary = I18n.t(
+        "ical.summary", facility: facility.abbreviation, product: product.name)
+      e.description = I18n.t(
```

Comment:
We might not be at the point yet, but it's looking like it might be worth extracting some private methods soon.